addons/website/static/src/snippets/s_image_gallery/options.js

```diff
@@ -110,8 +154,9 @@ options.registry.gallery = options.Class.extend({
     addImages: function (previewMode) {
         const $images = this.$('img');
         var $container = this.$('> .container, > .container-fluid, > .o_container_small');
-        const lastImage = _.last(this._getImages());
-        let index = lastImage ? this._getIndex(lastImage) : -1;
+        const mediaEls = this._getImages();
+        const lastMediaEl = mediaEls[mediaEls.length - 1];
+        let index = lastMediaEl ? this._getIndex(lastMediaEl) : -1;
```

Comment:
Not sure we need this change in stable :wink:

Reply:
But what if we replace an image by a video (considering that my fix properly keeps the `data-index` on it) and then add new images ? If the last media was the video, without this change, it will get the index of the last image and so the new images indexes will be wrong, no ? Since they will not start with the real last index, which is the video one :thinking:

Reply:
@sobo-odoo I must be blind but my point is that the previous code does the exact same thing as your new code (am I wrong?)

Before:
```javascript
const lastImage = _.last(this._getImages());
let index = lastImage ? this._getIndex(lastImage) : -1;
```
After:
```javascript
const mediaEls = this._getImages();
const lastMediaEl = mediaEls[mediaEls.length - 1];
let index = lastMediaEl ? this._getIndex(lastMediaEl) : -1;
```
In both cases, are not you getting the last item returned by `_getImages` and in both case calling `_getIndex` with it ?

Reply:
Aah my bad, I forgot it was still called `_getImages()` :smiling_face_with_tear: 

But I mentioned in my message before your review that a lot of changes were variables renaming, because it would allow to tell that there are not only images, which would avoid future issues, and were therefore not "cosmetic" changes

> About the big diff ( 😬) , it is mainly because there is a lot of variable renaming from imageEls, imageHolderEls, images, imgs... to mediaEls, mediaHolderEls. I know these could be considered as "cosmetic" changes but I discussed it with BSO before, and we concluded that it was well a fix, as it would allow to know that there are videos too, which would prevent doing something only image related in eventual future bug fixes


(it also allowed to remove jQuery code)

Does that mean that you do not agree with this type of changes ?

Reply:
It's ok :+1: Let's keep it :+1: